nodejs/week2/app.js

```diff
@@ -0,0 +1,99 @@
+const express = require("express");
+const app = express();
+const port = process.env.PORT || 3000;
+const documents = require("./documents.json");
+
+// Support parsing JSON requests
+app.use(express.json());
+
+app.get("/", (req, res) => {
+  res.send("This is a search engine");
+});
+
+// GET/search
+
+app.get("/search", (req, res) => {
+  try {
+    const containsQuery = req.query.q;
+
+    /* Object.values() method returns an array of 
+    a given object's own enumerable string-keyed property values. 
+    Some() method tests whether at least one element in the array 
+    passes the test implemented by the provided function. */
+
+    if (containsQuery) {
+      const matchQuery = documents.filter((item) =>
+        Object.values(item).some((element) =>
+          String(element).toLowerCase().includes(containsQuery)
+        )
+      );
+      res.send(matchQuery);
+    } else {
+      res.send(documents);
+    }
+  } catch (error) {
+    res.status(500).json({ message: "Something went wrong" });
+  }
+});
+
+// GET/documents/:id
+
+app.get("/documents/:id", (req, res) => {
+  try {
+    const matchId = documents.find(
+      (element) => element.id === parseInt(req.params.id)
+    );
+    if (matchId) {
+      res.json(matchId);
+    } else {
+      res.status(404).send(`ID ${req.params.id} not found`);
+    }
+  } catch (error) {
+    res.status(500).json({ error: "Something went wrong" });
+  }
+});
+
+// POST/search
+
+app.post("/search", async (req, res) => {
+  try {
+    const queryParameter = req.query.q;
+    const bodyParameter = req.body;
```

Comment:
In order to make search to work for bodyParameter you should assign it to req.body.fields;